src/server/protocol.ts

```diff
@@ -625,6 +626,30 @@ export interface GetMoveToRefactoringFileSuggestions extends Response {
     };
 }
 
+/**
+ * Request refactorings at a given position post pasting text from some other location.
+ */
+
+export interface GetPasteEditsRequest extends Request {
+    command: CommandTypes.GetPasteEdits;
+    arguments: GetPasteEditsRequestArgs;
+}
+
+export interface GetPasteEditsRequestArgs extends FileRequestArgs {
+    pastedText: string[];
+    pasteLocations: TextSpan[];
+    copiedFrom?: { file: string; range: TextSpan[]; };

```

Comment:
Last protocol naming suggestion 😅 

_Almost_ everything is consistent about calling a `TextSpan` a `span` or a `location` since internally, a range is something else. And the singular/plural mismatch here was really confusing when I was reading the code elsewhere.

```suggestion
    /** The source location of each `pastedText`. If present, the length of `spans` must be equal to the length of `pastedText`. */
    copiedFrom?: { file: string; spans: TextSpan[]; };
```

Additional comments would be good too.